### PR TITLE
Selenium exercises 1 to 13 - Completed (V1)

### PR DESCRIPTION
**Pull request by Vineet-Sawant**

### CODE REVIEW

File: selenium-exercise/1-chrome.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from selenium.webdriver import Chrome
+from selenium.webdriver.common.by import By
+from time import sleep
+from selenium.webdriver.chrome.service import Service as ChromeService
+
```
Comment (nshnt-xyz) [inline selenium-exercise/1-chrome.py:3]: You can create a class that initialises the drivers and then use it for all the exercises.

File: selenium-exercise/1-chrome.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+driver_path = r'C:\Users\sawan\Desktop\infuse\Practice-Vineet\selenium-exercise\drivers\chromedriver-win64\chromedriver.exe'
+chrome_service = ChromeService(executable_path = driver_path)
+
+driver = Chrome(service=chrome_service);
+
+driver.maximize_window()
+sleep(2)
```
Comment (nshnt-xyz) [inline selenium-exercise/1-chrome.py]: Don't use sleep.

File: selenium-exercise/2.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+driver_path = r'C:\Users\sawan\Desktop\infuse\Practice-Vineet\selenium-exercise\drivers\chromedriver-win64\chromedriver.exe'
+chrome_service = ChromeService(executable_path=driver_path)
+
+
+try:
+    driver = Chrome(service=chrome_service)
+    driver.get('https://www.google.com')
+    title = driver.title
+    print(f"Title: {title}")
+    assert title
```
Comment (nshnt-xyz) [inline selenium-exercise/2.py]: Is assert needed here?

File: selenium-exercise/4.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+try:
+    chrome_service = ChromeService(executable_path=driver_path)
+    driver = Chrome(service=chrome_service)
+    #actions = ActionChains(driver)
+    
+    driver.get('https://demo.automationtesting.in/Register.html')
+    driver.maximize_window()
+
+    radio_btn = driver.find_element(By.CSS_SELECTOR, value='#basicBootstrapForm > div:nth-child(5) > div > label:nth-child(1) > input')
```
Comment (nshnt-xyz) [inline selenium-exercise/4.py]: Use variables to store identifiers before using them in methods. This applies to all exercises.

File: selenium-exercise/4.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    check_box_1 = driver.find_element(By.ID, value="checkbox2")
```
Comment (nshnt-xyz) [inline selenium-exercise/4.py:13]: Use methods to do operations. This applies to all exercises.

File: selenium-exercise/5.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    driver = Chrome(service=chrome_service)
+    
+    
+    driver.get('https://www.yahoo.com/')
+    driver.maximize_window()
+    text_box = driver.find_element(by=By.ID, value="ybar-sbq")
+    submit_button = driver.find_element(by=By.ID, value="ybar-search")
+
+    text_box.send_keys("Nutty Putty Cave")
+    sleep(2)
```
Comment (nshnt-xyz) [inline selenium-exercise/5.py]: Don't use sleeps.

File: selenium-exercise/7.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+from selenium.webdriver.chrome.service import Service as ChromeService
+from selenium.common.exceptions import WebDriverException
+from selenium.webdriver.support.ui import Select
+
+from time import sleep
+
+driver_path = r'C:\Users\sawan\Desktop\infuse\Practice-Vineet\selenium-exercise\drivers\chromedriver-win64\chromedriver.exe'
+def select_ith_item_from_dropdown(i):
+    try:
+        chrome_service = ChromeService(executable_path=driver_path)
```
Comment (nshnt-xyz) [inline selenium-exercise/7.py]: This function is doing quite a lot of things, modularise it so its more readable.

File: selenium-exercise/8.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+try:
+    chrome_service = ChromeService(executable_path=driver_path)
+    driver = Chrome(service=chrome_service)
+    
+    
+    driver.get('https://www.globalsqa.com/demo-site/sliders/#Steps')
+    driver.maximize_window()
+
+    iframe = driver.find_element(By.XPATH, value='/html/body/div[1]/div[1]/div[2]/div/div/div[2]/div/div/div[3]/p/iframe')
```
Comment (nshnt-xyz) [inline selenium-exercise/8.py]: See if you can use anything other than XPATH.

File: selenium-exercise/8.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    driver.switch_to.frame(iframe)
+
+    slider = driver.find_element(By.ID, value='slider')
+    handler = slider.find_element(By.CLASS_NAME, value='ui-slider-handle')
+
+    action = ActionChains(driver)
+    action.click_and_hold(handler).move_by_offset(100, 0).release().perform()
```
Comment (nshnt-xyz) [inline selenium-exercise/8.py:19]: You can click on the slider button and then use arrow keys to do this. It'll be much cleaner.

File: selenium-exercise/12.py [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        remove_button.click()
+    except IndexError as e:
+        print(e)
+       
+def filter_products(driver):
+    header_section = driver.find_element(By.CLASS_NAME, 'header_secondary_container')
+    filter_options = header_section.find_elements(By.TAG_NAME, 'option')
+    filter_options[2].click()
+    # try:
+    #     dropdown = driver.find_element(By.CLASS_NAME, 'product_sort_container')
```
Comment (nshnt-xyz) [inline selenium-exercise/12.py]: Remove commented out code.

File: selenium-exercise/13.py [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+from selenium.webdriver.chrome.service import Service
+from selenium.webdriver.common.by import By
+from time import sleep
+ 
+chrome_driver_path = r'C:\Users\sawan\Desktop\infuse\Practice-Vineet\selenium-exercise\drivers\chromedriver-win64\chromedriver.exe'
+
+edge_driver_path = r'C:\Users\sawan\Desktop\infuse\Practice-Vineet\selenium-exercise\drivers\edgedriver_win64\msedgedriver.exe'
+browser=""
+ 
+def chrome_setUp():
```
Comment (nshnt-xyz) [inline selenium-exercise/13.py]: This isnt the right way to name methods, python uses snake case.

**Review by nshnt-xyz — CHANGES_REQUESTED** [review]
Modularise the code so that it is readable and reusable.
Use OOP concepts.
Use proper naming throughout the exercises.

File: selenium-exercise/1-chrome.py (hunk 1 above)
Comment (Vineet-Sawant) [reply selenium-exercise/1-chrome.py:3]: Done ✅

File: selenium-exercise/1-chrome.py (hunk 2 above)
Comment (Vineet-Sawant) [reply selenium-exercise/1-chrome.py]: Done ✅

File: selenium-exercise/2.py (hunk 3 above)
Comment (Vineet-Sawant) [reply selenium-exercise/2.py]: Done ✅

File: selenium-exercise/4.py (hunk 4 above)
Comment (Vineet-Sawant) [reply selenium-exercise/4.py]: Done ✅

File: selenium-exercise/5.py (hunk 6 above)
Comment (Vineet-Sawant) [reply selenium-exercise/5.py]: Done ✅

**Pull request merged**